totalRP3/modules/map/MapDataProvider.lua

```diff
@@ -0,0 +1,62 @@
+----------------------------------------------------------------------------------
+--- Total RP 3
+--- World map data provider
+---	---------------------------------------------------------------------------
+--- Copyright 2018 Renaud "Ellypse" Parize <ellypse@totalrp3.info> @EllypseCelwe
+---
+---	Licensed under the Apache License, Version 2.0 (the "License");
+---	you may not use this file except in compliance with the License.
+---	You may obtain a copy of the License at
+---
+---		http://www.apache.org/licenses/LICENSE-2.0
+---
+---	Unless required by applicable law or agreed to in writing, software
+---	distributed under the License is distributed on an "AS IS" BASIS,
+---	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+---	See the License for the specific language governing permissions and
+---	limitations under the License.
+----------------------------------------------------------------------------------
+
+---@type TRP3_API
+local _, TRP3_API = ...;
+local Ellyb = Ellyb(...);
+
+local pairs = pairs;
+
+---@type MapCanvasDataProviderMixin|{GetMap:fun():MapCanvasMixin}
+local TRP3_MapDataProvider = CreateFromMixins(MapCanvasDataProviderMixin);
+
+function TRP3_MapDataProvider:RemoveAllData()
+	self:GetMap():RemoveAllPinsByTemplate(self.pinTemplate);
+end
+
+function TRP3_MapDataProvider:RefreshAllData(fromOnShow, ...)
+	self:RemoveAllData();
+
+	if not self.data then return end
+
+	for _, POIInfo in pairs(self.data) do
```

Comment:
Casing-wise this should be "poiInfo"?

Reply:
Hmm, I guess, but I don't like it. It's debatable, but POI being an acronym I like it better as uppercase.

Reply:
After more reflexion, I think it's better to go with your suggestion as it follows the naming convention of Blizzard, which is something I try to do whenever possible.